In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [2]:
df = pd.read_csv("G:\Letsupgrade\Day 7\general_data.csv")

In [3]:
df.head()

Age Attrition     BusinessTravel              Department  DistanceFromHome  \
0   51        No      Travel_Rarely                   Sales                 6   
1   31       Yes  Travel_Frequently  Research & Development                10   
2   32        No  Travel_Frequently  Research & Development                17   
3   38        No         Non-Travel  Research & Development                 2   
4   32        No      Travel_Rarely  Research & Development                10   

   Education EducationField  EmployeeCount  EmployeeID  Gender  ...  \
0          2  Life Sciences              1           1  Female  ...   
1          1  Life Sciences              1           2  Female  ...   
2          4          Other              1           3    Male  ...   
3          5  Life Sciences              1           4    Male  ...   
4          1        Medical              1           5    Male  ...   

   NumCompaniesWorked Over18 PercentSalaryHike  StandardHours  \
0                 1.0      Y                11              8   
1                 0.0      Y                23              8   
2                 1.0      Y                15              8   
3                 3.0      Y                11              8   
4                 4.0      Y                12              8   

   StockOptionLevel TotalWorkingYears  TrainingTimesLastYear  YearsAtCompany  \
0                 0               1.0                      6               1   
1                 1               6.0                      3               5   
2                 3               5.0                      2               5   
3                 3              13.0                      5               8   
4                 2               9.0                      2               6   

   YearsSinceLastPromotion  YearsWithCurrManager  
0                        0                     0  
1                        1                     4  
2                        0                     3  
3                        7                     5  
4                        0                     4  

[5 rows x 24 columns]

In [4]:
df.drop_duplicates()

Age Attrition     BusinessTravel              Department  \
0      51        No      Travel_Rarely                   Sales   
1      31       Yes  Travel_Frequently  Research & Development   
2      32        No  Travel_Frequently  Research & Development   
3      38        No         Non-Travel  Research & Development   
4      32        No      Travel_Rarely  Research & Development   
5      46        No      Travel_Rarely  Research & Development   
6      28       Yes      Travel_Rarely  Research & Development   
7      29        No      Travel_Rarely  Research & Development   
8      31        No      Travel_Rarely  Research & Development   
9      25        No         Non-Travel  Research & Development   
10     45        No      Travel_Rarely  Research & Development   
11     36        No      Travel_Rarely  Research & Development   
12     55        No      Travel_Rarely  Research & Development   
13     47       Yes         Non-Travel  Research & Development   
14     28        No      Travel_Rarely  Research & Development   
15     37        No      Travel_Rarely  Research & Development   
16     21        No      Travel_Rarely  Research & Development   
17     37        No         Non-Travel  Research & Development   
18     35        No      Travel_Rarely                   Sales   
19     38        No      Travel_Rarely  Research & Development   
20     26        No  Travel_Frequently  Research & Development   
21     50        No      Travel_Rarely                   Sales   
22     53        No      Travel_Rarely  Research & Development   
23     42        No      Travel_Rarely  Research & Development   
24     29        No  Travel_Frequently  Research & Development   
25     55        No      Travel_Rarely  Research & Development   
26     26        No  Travel_Frequently  Research & Development   
27     37        No      Travel_Rarely                   Sales   
28     44       Yes  Travel_Frequently  Research & Development   
29     38        No      Travel_Rarely                   Sales   
...   ...       ...                ...                     ...   
4380   35        No  Travel_Frequently  Research & Development   
4381   29       Yes      Travel_Rarely  Research & Development   
4382   52        No      Travel_Rarely  Research & Development   
4383   42        No      Travel_Rarely  Research & Development   
4384   59        No      Travel_Rarely  Research & Development   
4385   50        No      Travel_Rarely  Research & Development   
4386   33       Yes      Travel_Rarely                   Sales   
4387   43        No      Travel_Rarely                   Sales   
4388   33       Yes      Travel_Rarely                   Sales   
4389   52        No         Non-Travel  Research & Development   
4390   32        No      Travel_Rarely         Human Resources   
4391   32       Yes      Travel_Rarely                   Sales   
4392   39        No      Travel_Rarely                   Sales   
4393   32        No         Non-Travel                   Sales   
4394   41        No      Travel_Rarely                   Sales   
4395   40        No      Travel_Rarely  Research & Development   
4396   45        No      Travel_Rarely  Research & Development   
4397   31        No  Travel_Frequently  Research & Development   
4398   33        No      Travel_Rarely  Research & Development   
4399   34        No      Travel_Rarely  Research & Development   
4400   37        No      Travel_Rarely  Research & Development   
4401   45        No  Travel_Frequently                   Sales   
4402   37       Yes  Travel_Frequently                   Sales   
4403   39        No  Travel_Frequently  Research & Development   
4404   29        No      Travel_Rarely                   Sales   
4405   42        No      Travel_Rarely  Research & Development   
4406   29        No      Travel_Rarely  Research & Development   
4407   25        No      Travel_Rarely  Research & Development   
4408   42        No      Travel_Rarely        

In [8]:
dataYes = df[df['Attrition'] == "Yes"]
dataNo = df[df['Attrition'] == "No"]

# Employees who left the company:

In [7]:
dataYes.describe()

Age  DistanceFromHome   Education  EmployeeCount   EmployeeID  \
count  711.000000        711.000000  711.000000          711.0   711.000000   
mean    33.607595          9.012658    2.877637            1.0  2191.767932   
std      9.675693          7.772368    1.014233            0.0  1270.136654   
min     18.000000          1.000000    1.000000            1.0     2.000000   
25%     28.000000          2.000000    2.000000            1.0  1072.000000   
50%     32.000000          7.000000    3.000000            1.0  2165.000000   
75%     39.000000         15.000000    4.000000            1.0  3341.000000   
max     58.000000         29.000000    5.000000            1.0  4403.000000   

         JobLevel  MonthlyIncome  NumCompaniesWorked  PercentSalaryHike  \
count  711.000000     711.000000          707.000000         711.000000   
mean     2.037975   61682.616034            2.936351          15.481013   
std      1.057485   44792.067695            2.678774           3.775289   
min      1.000000   10090.000000            0.000000          11.000000   
25%      1.000000   28440.000000            1.000000          12.000000   
50%      2.000000   49080.000000            1.000000          14.000000   
75%      2.000000   71040.000000            5.000000          18.000000   
max      5.000000  198590.000000            9.000000          25.000000   

       StandardHours  StockOptionLevel  TotalWorkingYears  \
count          711.0        711.000000         709.000000   
mean             8.0          0.780591           8.255289   
std              0.0          0.858899           7.164018   
min              8.0          0.000000           0.000000   
25%              8.0          0.000000           3.000000   
50%              8.0          1.000000           7.000000   
75%              8.0          1.000000          10.000000   
max              8.0          3.000000          40.000000   

       TrainingTimesLastYear  YearsAtCompany  YearsSinceLastPromotion  \
count             711.000000      711.000000               711.000000   
mean                2.654008        5.130802                 1.945148   
std                 1.154834        5.941598                 3.148633   
min                 0.000000        0.000000                 0.000000   
25%                 2.000000        1.000000                 0.000000   
50%                 3.000000        3.000000                 1.000000   
75%                 3.000000        7.000000                 2.000000   
max                 6.000000       40.000000                15.000000   

       YearsWithCurrManager  
count            711.000000  
mean               2.852321  
std                3.138918  
min                0.000000  
25%                0.000000  
50%                2.000000  
75%                5.000000  
max               14.000000

In [25]:
dataYes.sample(50).describe()

Age  DistanceFromHome  Education  EmployeeCount   EmployeeID  \
count  50.000000         50.000000  50.000000           50.0    50.000000   
mean   33.140000          8.220000   2.740000            1.0  2306.500000   
std    10.012258          7.887707   1.121406            0.0  1242.579382   
min    18.000000          1.000000   1.000000            1.0    31.000000   
25%    28.000000          2.000000   2.000000            1.0  1363.000000   
50%    31.000000          5.500000   3.000000            1.0  2507.000000   
75%    37.750000         10.000000   3.000000            1.0  3372.000000   
max    58.000000         29.000000   5.000000            1.0  4279.000000   

        JobLevel  MonthlyIncome  NumCompaniesWorked  PercentSalaryHike  \
count  50.000000      50.000000           50.000000           50.00000   
mean    1.980000   64610.400000            3.080000           14.72000   
std     0.979171   48495.015648            2.898557            3.22009   
min     1.000000   11290.000000            0.000000           11.00000   
25%     1.000000   27410.000000            1.000000           12.00000   
50%     2.000000   45585.000000            1.000000           13.00000   
75%     2.000000   87770.000000            5.750000           17.00000   
max     5.000000  191900.000000            9.000000           22.00000   

       StandardHours  StockOptionLevel  TotalWorkingYears  \
count           50.0         50.000000          50.000000   
mean             8.0          0.760000           6.800000   
std              0.0          0.846602           6.490181   
min              8.0          0.000000           0.000000   
25%              8.0          0.000000           3.000000   
50%              8.0          1.000000           6.000000   
75%              8.0          1.000000           9.000000   
max              8.0          3.000000          40.000000   

       TrainingTimesLastYear  YearsAtCompany  YearsSinceLastPromotion  \
count              50.000000       50.000000                50.000000   
mean                2.860000        4.100000                 1.240000   
std                 1.277913        6.165242                 2.722244   
min                 0.000000        0.000000                 0.000000   
25%                 2.000000        1.000000                 0.000000   
50%                 3.000000        2.000000                 0.000000   
75%                 3.000000        6.000000                 1.750000   
max                 6.000000       40.000000                15.000000   

       YearsWithCurrManager  
count             50.000000  
mean               2.280000  
std                2.748209  
min                0.000000  
25%                0.000000  
50%                1.000000  
75%                3.000000  
max                8.000000

# Employees who are still  the part of company:

In [9]:
dataNo.describe()

Age  DistanceFromHome    Education  EmployeeCount   EmployeeID  \
count  3699.000000       3699.000000  3699.000000         3699.0  3699.000000   
mean     37.561233          9.227088     2.919708            1.0  2208.139497   
std       8.885956          8.167978     1.025784            0.0  1273.944391   
min      18.000000          1.000000     1.000000            1.0     1.000000   
25%      31.000000          2.000000     2.000000            1.0  1110.500000   
50%      36.000000          7.000000     3.000000            1.0  2213.000000   
75%      43.000000         14.000000     4.000000            1.0  3304.500000   
max      60.000000         29.000000     5.000000            1.0  4410.000000   

          JobLevel  MonthlyIncome  NumCompaniesWorked  PercentSalaryHike  \
count  3699.000000    3699.000000         3684.000000        3699.000000   
mean      2.068938   65672.595296            2.648480          15.157340   
std       1.115967   47472.814021            2.460537           3.634551   
min       1.000000   10510.000000            0.000000          11.000000   
25%       1.000000   29360.000000            1.000000          12.000000   
50%       2.000000   49300.000000            2.000000          14.000000   
75%       3.000000   86060.000000            4.000000          18.000000   
max       5.000000  199990.000000            9.000000          25.000000   

       StandardHours  StockOptionLevel  TotalWorkingYears  \
count         3699.0       3699.000000        3692.000000   
mean             8.0          0.796431          11.860780   
std              0.0          0.850621           7.762536   
min              8.0          0.000000           0.000000   
25%              8.0          0.000000           6.000000   
50%              8.0          1.000000          10.000000   
75%              8.0          1.000000          16.000000   
max              8.0          3.000000          38.000000   

       TrainingTimesLastYear  YearsAtCompany  YearsSinceLastPromotion  \
count            3699.000000     3699.000000              3699.000000   
mean                2.827251        7.369019                 2.234388   
std                 1.311493        6.094649                 3.233887   
min                 0.000000        0.000000                 0.000000   
25%                 2.000000        3.000000                 0.000000   
50%                 3.000000        6.000000                 1.000000   
75%                 3.000000       10.000000                 3.000000   
max                 6.000000       37.000000                15.000000   

       YearsWithCurrManager  
count           3699.000000  
mean               4.367397  
std                3.593144  
min                0.000000  
25%                2.000000  
50%                3.000000  
75%                7.000000  
max               17.000000

In [26]:
dataNo.sample(50).describe()

Age  DistanceFromHome  Education  EmployeeCount   EmployeeID  \
count  50.000000          50.00000  50.000000           50.0    50.000000   
mean   37.460000           8.28000   2.960000            1.0  2378.160000   
std     9.221116           7.97711   1.029365            0.0  1246.865054   
min    23.000000           1.00000   1.000000            1.0    80.000000   
25%    30.250000           2.00000   2.250000            1.0  1278.000000   
50%    37.000000           5.50000   3.000000            1.0  2473.500000   
75%    43.750000          12.00000   4.000000            1.0  3315.250000   
max    58.000000          29.00000   5.000000            1.0  4397.000000   

        JobLevel  MonthlyIncome  NumCompaniesWorked  PercentSalaryHike  \
count  50.000000      50.000000           50.000000          50.000000   
mean    2.200000   61067.600000            2.160000          15.200000   
std     1.228904   45170.121358            1.765428           3.625308   
min     1.000000   16110.000000            0.000000          11.000000   
25%     1.000000   29387.500000            1.000000          12.250000   
50%     2.000000   43210.000000            2.000000          14.000000   
75%     3.000000   88275.000000            3.750000          17.750000   
max     5.000000  196260.000000            7.000000          23.000000   

       StandardHours  StockOptionLevel  TotalWorkingYears  \
count           50.0         50.000000          50.000000   
mean             8.0          0.800000          11.680000   
std              0.0          0.808122           8.560565   
min              8.0          0.000000           1.000000   
25%              8.0          0.000000           7.000000   
50%              8.0          1.000000          10.000000   
75%              8.0          1.000000          14.500000   
max              8.0          3.000000          35.000000   

       TrainingTimesLastYear  YearsAtCompany  YearsSinceLastPromotion  \
count              50.000000       50.000000                 50.00000   
mean                2.500000        6.560000                  2.00000   
std                 1.233048        4.376001                  2.56348   
min                 0.000000        1.000000                  0.00000   
25%                 2.000000        3.000000                  0.00000   
50%                 2.000000        5.500000                  1.00000   
75%                 3.000000        9.000000                  2.00000   
max                 5.000000       24.000000                  9.00000   

       YearsWithCurrManager  
count             50.000000  
mean               4.100000  
std                3.118411  
min                0.000000  
25%                2.000000  
50%                3.000000  
75%                7.000000  
max               13.000000

# Employees who are part of the company more than equal to 4 years

In [13]:
df1 = dataYes.loc[df['YearsAtCompany'] >=4 ]
df1.describe()

Age  DistanceFromHome   Education  EmployeeCount   EmployeeID  \
count  345.000000        345.000000  345.000000          345.0   345.000000   
mean    37.417391          9.608696    2.860870            1.0  2222.478261   
std      9.324557          8.096829    1.004817            0.0  1272.066114   
min     23.000000          1.000000    1.000000            1.0     2.000000   
25%     30.000000          2.000000    2.000000            1.0  1097.000000   
50%     35.000000          7.000000    3.000000            1.0  2198.000000   
75%     44.000000         15.000000    4.000000            1.0  3350.000000   
max     58.000000         29.000000    5.000000            1.0  4389.000000   

         JobLevel  MonthlyIncome  NumCompaniesWorked  PercentSalaryHike  \
count  345.000000     345.000000          343.000000         345.000000   
mean     2.043478   57994.695652            2.997085          15.643478   
std      1.059766   42194.295361            2.759882           3.997509   
min      1.000000   10090.000000            0.000000          11.000000   
25%      1.000000   27430.000000            1.000000          13.000000   
50%      2.000000   46800.000000            1.000000          14.000000   
75%      2.000000   67550.000000            5.000000          19.000000   
max      5.000000  196270.000000            9.000000          25.000000   

       StandardHours  StockOptionLevel  TotalWorkingYears  \
count          345.0        345.000000         344.000000   
mean             8.0          0.834783          11.863372   
std              0.0          0.914328           7.309999   
min              8.0          0.000000           4.000000   
25%              8.0          0.000000           7.000000   
50%              8.0          1.000000          10.000000   
75%              8.0          1.000000          14.000000   
max              8.0          3.000000          40.000000   

       TrainingTimesLastYear  YearsAtCompany  YearsSinceLastPromotion  \
count             345.000000      345.000000               345.000000   
mean                2.391304        9.069565                 3.408696   
std                 1.171580        6.461498                 3.948570   
min                 0.000000        4.000000                 0.000000   
25%                 2.000000        5.000000                 0.000000   
50%                 2.000000        7.000000                 1.000000   
75%                 3.000000       10.000000                 7.000000   
max                 6.000000       40.000000                15.000000   

       YearsWithCurrManager  
count            345.000000  
mean               5.130435  
std                3.050510  
min                0.000000  
25%                3.000000  
50%                5.000000  
75%                7.000000  
max               14.000000

In [14]:
df1.sample(50).describe()

Age  DistanceFromHome  Education  EmployeeCount   EmployeeID  \
count  50.00000         50.000000  50.000000           50.0    50.000000   
mean   37.02000         11.160000   2.980000            1.0  2346.160000   
std     8.66023          8.772313   1.039819            0.0  1319.511402   
min    24.00000          1.000000   1.000000            1.0   157.000000   
25%    31.00000          2.250000   2.000000            1.0  1357.250000   
50%    35.50000         10.000000   3.000000            1.0  2284.500000   
75%    42.00000         18.000000   4.000000            1.0  3508.750000   
max    58.00000         29.000000   5.000000            1.0  4382.000000   

        JobLevel  MonthlyIncome  NumCompaniesWorked  PercentSalaryHike  \
count  50.000000      50.000000           50.000000          50.000000   
mean    2.060000   50567.800000            2.660000          16.240000   
std     1.167764   35041.476316            2.781939           3.684274   
min     1.000000   15550.000000            0.000000          11.000000   
25%     1.000000   23212.500000            1.000000          13.000000   
50%     2.000000   42475.000000            1.000000          15.500000   
75%     2.750000   61565.000000            4.000000          19.000000   
max     5.000000  167990.000000            9.000000          24.000000   

       StandardHours  StockOptionLevel  TotalWorkingYears  \
count           50.0         50.000000          50.000000   
mean             8.0          0.840000          11.980000   
std              0.0          0.997139           7.468273   
min              8.0          0.000000           4.000000   
25%              8.0          0.000000           6.000000   
50%              8.0          1.000000          10.000000   
75%              8.0          1.000000          15.000000   
max              8.0          3.000000          40.000000   

       TrainingTimesLastYear  YearsAtCompany  YearsSinceLastPromotion  \
count              50.000000         50.0000                50.000000   
mean                2.400000          9.8600                 4.020000   
std                 1.160577          6.5528                 4.537778   
min                 0.000000          4.0000                 0.000000   
25%                 2.000000          5.0000                 0.000000   
50%                 2.500000          7.0000                 1.500000   
75%                 3.000000         10.0000                 7.000000   
max                 6.000000         31.0000                15.000000   

       YearsWithCurrManager  
count             50.000000  
mean               5.500000  
std                3.441938  
min                0.000000  
25%                3.000000  
50%                6.000000  
75%                7.750000  
max               14.000000

# Salary Hike

In [17]:
dfSalary =  dataYes.loc[df['PercentSalaryHike'] >= 20]
dfSalary.describe()

Age  DistanceFromHome   Education  EmployeeCount   EmployeeID  \
count  123.000000        123.000000  123.000000          123.0   123.000000   
mean    33.365854         10.000000    2.780488            1.0  2216.804878   
std      9.563651          7.355159    1.097961            0.0  1304.208281   
min     19.000000          1.000000    1.000000            1.0     2.000000   
25%     28.000000          5.000000    2.000000            1.0  1283.000000   
50%     33.000000          8.000000    3.000000            1.0  2249.000000   
75%     36.000000         15.000000    4.000000            1.0  3149.000000   
max     56.000000         28.000000    4.000000            1.0  4387.000000   

         JobLevel  MonthlyIncome  NumCompaniesWorked  PercentSalaryHike  \
count  123.000000     123.000000           123.00000         123.000000   
mean     2.000000   54536.341463             3.02439          21.975610   
std      1.086429   40041.504456             3.04465           1.621896   
min      1.000000   19040.000000             0.00000          20.000000   
25%      1.000000   25930.000000             1.00000          21.000000   
50%      2.000000   40780.000000             1.00000          22.000000   
75%      2.000000   65000.000000             5.00000          23.000000   
max      5.000000  192720.000000             9.00000          25.000000   

       StandardHours  StockOptionLevel  TotalWorkingYears  \
count          123.0        123.000000         123.000000   
mean             8.0          0.512195           7.317073   
std              0.0          0.669791           6.199309   
min              8.0          0.000000           0.000000   
25%              8.0          0.000000           4.000000   
50%              8.0          0.000000           6.000000   
75%              8.0          1.000000           9.000000   
max              8.0          3.000000          34.000000   

       TrainingTimesLastYear  YearsAtCompany  YearsSinceLastPromotion  \
count             123.000000      123.000000               123.000000   
mean                2.829268        4.829268                 1.585366   
std                 1.150248        5.358779                 2.206276   
min                 0.000000        0.000000                 0.000000   
25%                 2.000000        2.000000                 0.000000   
50%                 3.000000        4.000000                 1.000000   
75%                 3.000000        7.000000                 2.000000   
max                 5.000000       33.000000                 9.000000   

       YearsWithCurrManager  
count             123.00000  
mean                3.04878  
std                 2.82220  
min                 0.00000  
25%                 0.00000  
50%                 2.00000  
75%                 7.00000  
max                 9.00000

In [27]:
dfSalary.sample(50).describe()

Age  DistanceFromHome  Education  EmployeeCount   EmployeeID  \
count  50.000000         50.000000  50.000000           50.0    50.000000   
mean   35.320000          9.660000   2.760000            1.0  2459.000000   
std     9.459775          6.793184   1.134973            0.0  1179.954513   
min    19.000000          1.000000   1.000000            1.0   179.000000   
25%    31.000000          4.250000   2.000000            1.0  1503.000000   
50%    34.000000         10.000000   3.000000            1.0  2601.500000   
75%    40.750000         15.000000   4.000000            1.0  3280.250000   
max    56.000000         28.000000   4.000000            1.0  4382.000000   

        JobLevel  MonthlyIncome  NumCompaniesWorked  PercentSalaryHike  \
count  50.000000      50.000000           50.000000          50.000000   
mean    1.960000   54533.600000            3.060000          22.100000   
std     1.049003   37912.933271            2.958281           1.693324   
min     1.000000   19040.000000            0.000000          20.000000   
25%     1.000000   26762.500000            1.000000          21.000000   
50%     2.000000   41880.000000            1.500000          22.000000   
75%     2.000000   66520.000000            5.000000          23.000000   
max     5.000000  192720.000000            9.000000          25.000000   

       StandardHours  StockOptionLevel  TotalWorkingYears  \
count           50.0           50.0000          50.000000   
mean             8.0            0.5600           8.200000   
std              0.0            0.7329           5.900536   
min              8.0            0.0000           0.000000   
25%              8.0            0.0000           4.000000   
50%              8.0            0.0000           8.000000   
75%              8.0            1.0000          10.750000   
max              8.0            3.0000          34.000000   

       TrainingTimesLastYear  YearsAtCompany  YearsSinceLastPromotion  \
count              50.000000       50.000000                50.000000   
mean                2.640000        5.320000                 1.920000   
std                 1.173865        5.007912                 2.546226   
min                 0.000000        0.000000                 0.000000   
25%                 2.000000        2.000000                 0.000000   
50%                 3.000000        5.000000                 1.000000   
75%                 3.000000        7.000000                 2.000000   
max                 5.000000       33.000000                 9.000000   

       YearsWithCurrManager  
count             50.000000  
mean               3.500000  
std                2.866058  
min                0.000000  
25%                2.000000  
50%                2.500000  
75%                7.000000  
max                9.000000

# Employees who worked in more than 5 company

In [20]:
df2 =  dataYes.loc[df['NumCompaniesWorked'] >= 5]
df2.describe()

Age  DistanceFromHome   Education  EmployeeCount   EmployeeID  \
count  199.000000        199.000000  199.000000          199.0   199.000000   
mean    36.994975          8.552764    3.085427            1.0  2208.562814   
std      9.713833          7.114928    0.946945            0.0  1279.089447   
min     22.000000          1.000000    1.000000            1.0    66.000000   
25%     29.000000          2.000000    3.000000            1.0  1162.000000   
50%     34.000000          7.000000    3.000000            1.0  2179.000000   
75%     43.000000         11.500000    4.000000            1.0  3279.500000   
max     58.000000         28.000000    5.000000            1.0  4403.000000   

         JobLevel  MonthlyIncome  NumCompaniesWorked  PercentSalaryHike  \
count  199.000000     199.000000          199.000000         199.000000   
mean     2.015075   58937.135678            6.748744          15.844221   
std      0.912746   46344.372070            1.387966           4.134220   
min      1.000000   10090.000000            5.000000          11.000000   
25%      1.000000   27410.000000            6.000000          12.000000   
50%      2.000000   45220.000000            7.000000          15.000000   
75%      2.000000   67120.000000            8.000000          19.000000   
max      5.000000  196270.000000            9.000000          25.000000   

       StandardHours  StockOptionLevel  TotalWorkingYears  \
count          199.0        199.000000         199.000000   
mean             8.0          0.738693          10.849246   
std              0.0          0.859916           7.233996   
min              8.0          0.000000           3.000000   
25%              8.0          0.000000           6.000000   
50%              8.0          1.000000           8.000000   
75%              8.0          1.000000          15.000000   
max              8.0          3.000000          40.000000   

       TrainingTimesLastYear  YearsAtCompany  YearsSinceLastPromotion  \
count             199.000000      199.000000               199.000000   
mean                2.562814        5.306533                 1.989950   
std                 1.203876        5.874230                 3.231771   
min                 0.000000        0.000000                 0.000000   
25%                 2.000000        2.000000                 0.000000   
50%                 2.000000        4.000000                 1.000000   
75%                 3.000000        6.000000                 2.000000   
max                 6.000000       31.000000                14.000000   

       YearsWithCurrManager  
count            199.000000  
mean               2.924623  
std                2.737570  
min                0.000000  
25%                0.500000  
50%                2.000000  
75%                4.000000  
max               10.000000

In [21]:
df2.sample(50).describe()

Age  DistanceFromHome  Education  EmployeeCount   EmployeeID  \
count  50.000000         50.000000  50.000000           50.0    50.000000   
mean   37.280000          9.400000   3.300000            1.0  2161.320000   
std    10.339226          7.588929   0.952976            0.0  1153.475112   
min    24.000000          1.000000   1.000000            1.0    66.000000   
25%    29.000000          3.000000   3.000000            1.0  1363.000000   
50%    33.000000          8.000000   3.000000            1.0  2231.500000   
75%    43.000000         11.750000   4.000000            1.0  3097.250000   
max    58.000000         28.000000   5.000000            1.0  4073.000000   

        JobLevel  MonthlyIncome  NumCompaniesWorked  PercentSalaryHike  \
count  50.000000      50.000000           50.000000           50.00000   
mean    2.200000   62558.800000            6.520000           15.70000   
std     1.030158   46050.699507            1.388583            4.03682   
min     1.000000   10090.000000            5.000000           11.00000   
25%     1.250000   30720.000000            5.000000           12.00000   
50%     2.000000   48450.000000            6.000000           15.00000   
75%     3.000000   73560.000000            7.000000           17.00000   
max     5.000000  195660.000000            9.000000           25.00000   

       StandardHours  StockOptionLevel  TotalWorkingYears  \
count           50.0         50.000000          50.000000   
mean             8.0          0.720000          10.840000   
std              0.0          0.858095           8.298906   
min              8.0          0.000000           3.000000   
25%              8.0          0.000000           6.000000   
50%              8.0          1.000000           8.000000   
75%              8.0          1.000000          13.750000   
max              8.0          3.000000          40.000000   

       TrainingTimesLastYear  YearsAtCompany  YearsSinceLastPromotion  \
count              50.000000       50.000000                50.000000   
mean                2.520000        5.620000                 2.280000   
std                 1.110984        6.651346                 3.580218   
min                 0.000000        0.000000                 0.000000   
25%                 2.000000        2.000000                 0.000000   
50%                 3.000000        4.000000                 1.000000   
75%                 3.000000        6.750000                 2.000000   
max                 5.000000       31.000000                13.000000   

       YearsWithCurrManager  
count              50.00000  
mean                3.02000  
std                 2.83196  
min                 0.00000  
25%                 0.25000  
50%                 2.00000  
75%                 4.00000  
max                 9.00000

# Employees distance from home

In [22]:
dfDistance = dataYes.loc[df['DistanceFromHome']>=10]
dfDistance.describe()

Age  DistanceFromHome   Education  EmployeeCount   EmployeeID  \
count  270.000000        270.000000  270.000000          270.0   270.000000   
mean    34.211111         17.266667    2.922222            1.0  2172.888889   
std      9.727699          6.069854    1.004389            0.0  1270.676219   
min     19.000000         10.000000    1.000000            1.0     2.000000   
25%     28.000000         11.000000    2.000000            1.0  1049.500000   
50%     32.000000         16.000000    3.000000            1.0  2127.000000   
75%     41.000000         22.000000    4.000000            1.0  3295.500000   
max     58.000000         29.000000    5.000000            1.0  4392.000000   

         JobLevel  MonthlyIncome  NumCompaniesWorked  PercentSalaryHike  \
count  270.000000     270.000000          269.000000         270.000000   
mean     1.911111   58684.000000            2.825279          15.800000   
std      0.997891   44220.200067            2.640661           4.027415   
min      1.000000   15550.000000            0.000000          11.000000   
25%      1.000000   28100.000000            1.000000          12.000000   
50%      2.000000   44280.000000            1.000000          15.000000   
75%      2.000000   66960.000000            5.000000          19.000000   
max      5.000000  198590.000000            9.000000          25.000000   

       StandardHours  StockOptionLevel  TotalWorkingYears  \
count          270.0        270.000000         270.000000   
mean             8.0          0.777778           8.833333   
std              0.0          0.880745           7.416825   
min              8.0          0.000000           1.000000   
25%              8.0          0.000000           4.000000   
50%              8.0          1.000000           7.000000   
75%              8.0          1.000000          10.000000   
max              8.0          3.000000          40.000000   

       TrainingTimesLastYear  YearsAtCompany  YearsSinceLastPromotion  \
count             270.000000      270.000000               270.000000   
mean                2.733333        5.933333                 2.188889   
std                 1.238779        6.288244                 3.121675   
min                 0.000000        0.000000                 0.000000   
25%                 2.000000        2.000000                 0.000000   
50%                 3.000000        4.000000                 1.000000   
75%                 3.000000        8.000000                 3.000000   
max                 6.000000       33.000000                15.000000   

       YearsWithCurrManager  
count            270.000000  
mean               3.322222  
std                3.019741  
min                0.000000  
25%                0.000000  
50%                2.000000  
75%                7.000000  
max               11.000000

In [24]:
dfDistance.sample(50).describe()

Age  DistanceFromHome  Education  EmployeeCount   EmployeeID  \
count  50.000000         50.000000  50.000000           50.0    50.000000   
mean   36.940000         16.800000   3.000000            1.0  2252.320000   
std    10.254735          5.518059   0.989743            0.0  1299.506884   
min    19.000000         10.000000   1.000000            1.0   164.000000   
25%    29.000000         10.500000   2.000000            1.0  1116.750000   
50%    35.000000         17.500000   3.000000            1.0  2068.500000   
75%    44.000000         19.000000   4.000000            1.0  3418.250000   
max    58.000000         28.000000   4.000000            1.0  4358.000000   

       JobLevel  MonthlyIncome  NumCompaniesWorked  PercentSalaryHike  \
count  50.00000      50.000000           49.000000           50.00000   
mean    2.28000   52425.000000            2.836735           17.30000   
std     1.03095   43240.190123            2.830981            4.51415   
min     1.00000   15550.000000            0.000000           11.00000   
25%     2.00000   25675.000000            1.000000           13.00000   
50%     2.00000   43060.000000            1.000000           17.00000   
75%     2.00000   58530.000000            5.000000           21.00000   
max     5.00000  198590.000000            9.000000           25.00000   

       StandardHours  StockOptionLevel  TotalWorkingYears  \
count           50.0         50.000000           50.00000   
mean             8.0          0.880000           12.38000   
std              0.0          0.872248           10.35865   
min              8.0          0.000000            1.00000   
25%              8.0          0.000000            6.00000   
50%              8.0          1.000000           10.00000   
75%              8.0          1.000000           15.00000   
max              8.0          3.000000           40.00000   

       TrainingTimesLastYear  YearsAtCompany  YearsSinceLastPromotion  \
count              50.000000        50.00000                50.000000   
mean                2.680000         8.98000                 2.420000   
std                 0.998775         9.53295                 3.320253   
min                 1.000000         0.00000                 0.000000   
25%                 2.000000         2.25000                 0.000000   
50%                 3.000000         5.50000                 1.000000   
75%                 3.000000        10.00000                 2.750000   
max                 5.000000        33.00000                13.000000   

       YearsWithCurrManager  
count              50.00000  
mean                4.24000  
std                 3.51417  
min                 0.00000  
25%                 1.25000  
50%                 3.00000  
75%                 7.00000  
max                11.00000

# Hypothesis 1

# Null Hypothesis

The mean age of employees who has attrition is found to be 33.60 with standard deviation 9.67

# Alternative Hypothesis: 


The mean age of an employee who has attrite is not equal to 33
To determine if this is true, a sample of 50 attrite employees was taken
and their mean age was found to be 33.14

# Hypothesis 2

# Null Hypothesis:

The mean years at company of attrite employees was found to be
greater than or equal to 4 years was 9.06 with a standard deviation of
6.46.

# Alternative Hypothesis

The mean years at company of attrite employees is less than or
equal to 4 years.
In order to determine if this is true, a sample of 50 attrite
employees was taken at random and their mean years at the company
was found to be 8 years



# Hypothesis 3

# Null Hypothesis:
The mean percentage in salary hike of the attrite employees who
took training more than 4 times in the previous year is found to be
greater than or equal to 15%

# Alternative Hypothesis:
The mean percentage in salary hike of the attrite employees who
took training more than 4 times in the previous year is found to be less
than 15%.
In order to test this claim, a sample of 50 employees who have
attrite and took training more than 4 times in the previous year was
taken at random and their mean percentage in salary hike was found to
be 15.78%

# Hypothesis 4

# Null Hypothesis:
The mean salary of attrite employees who have worked in more
than 5 companies is found to be Rs.58,937 with a standard deviation of
46344.


# Alternative Hypothesis:
The mean salary of attrite employees who have worked in more
than 5 companies is not equal to Rs.58,937
In order to test this claim, a sample of 50 employees who have
attrite and worked in more than 5 companies was taken at random and
mean salary was found to be Rs 57,679

# Hypothesis 5

# Null Hypothesis:
Mean of total working years of attrite employees was found to less
than or equal to 10 years with a standard deviation of 7.16

# Alternative Hypothesis:
Mean of total working years of attrite employees is greater than 10
years.
In order to test this claim, a sample of 50 employees who have
attrite was taken at random and their mean working years was found to
be 8